### DISCUSSION TITLE
Ability to display multi-day and single-day event in 1 nova resource?

### DISCUSSION THREAD

**Discussion by pitchayakit** [wdelfuego/nova-calendar#34] [Q&A] [answered]
For example, I have 2 rows for some resources like
1. starts_at, end_at --> this row works properly
2. starts_at, null --> if end_at is null, should be considered for a single day but I didn't see how to display it on calendar.

https://github.com/wdelfuego/nova-calendar/blob/571673911c01978a2dfa44d4791f4e4e0f7068d3/resources/CalendarDataProvider.default.php#L39

Could you please suggest a solution?

**Comment by wdelfuego** [comment]
Tl;dr: it was a bug, fix released in version 1.6


The expected behavior is that this should work out of the box; if you supply property names for both a start and an end timestamp, the event should be shown as a single day event if the end timestamp is either `null` or on the same day as the start timestamp, and as a multi day event otherwise. Strange things might happen if the object has an end timestamp that lies before the start timestamp.

What is the behavior you're observing?

**Reply by pitchayakit** [reply]
As I tried if end timestamp is null, the event will not display in the calendar.
My setting is `SomeResource::class => ['starts_at', 'ends_at'], `
if I change to `SomeResource::class => starts_at`, event will display in calendat but not multi-days.

My observation is, is it possible to display multi-day and single-day for end timestamp as null in one resource?

**Reply by wdelfuego** [reply]
Not only should it be possible, it is the default expected behavior and should work out of the box, so if this doesn't work well it's a bug and I'll see if I can reproduce it. Thanks for reporting!

**Reply by wdelfuego** [reply]
I can confirm this is a bug, I'll release the fix asap. Thank you very much for bringing this to my attention.

**Reply by wdelfuego** [reply]
Bug is fixed, release 1.6 will be out on Monday at the latest.

**Reply by wdelfuego** [reply]
Created issue #35 for further handling this bug, closing this Discussion now

**Comment by wdelfuego** [comment, marked as answer]
Version 1.6 that fixes this bug has just been released
